### PR TITLE
Containers: Fix remaining test cases pulling from docker.io

### PR DESCRIPTION
**Pull request by jlausuch**
Affected tests:
- containers_3rd_party
- docker_compose
- registry

See failures: https://openqa.opensuse.org/tests/1492189

VR: TBD

### CODE REVIEW

**Comment by Vogtinator** [comment]
> VR: TBD

https://openqa.opensuse.org/tests/1492249

File: tests/containers/containers_3rd_party.pm [hunk 1]
```diff
@@ -64,8 +64,10 @@ sub run {
     my ($running_version, $sp, $host_distri) = get_os_release;
 
     # Define test images here
-    my @docker_images = ('alpine', 'opensuse/leap', 'opensuse/tumbleweed', 'debian', 'ubuntu', 'centos', 'fedora');
-    my @podman_images = ('alpine', 'opensuse/leap', 'opensuse/tumbleweed', 'debian', 'ubuntu', 'centos', 'fedora');
+    my $ex_reg        = get_var('REGISTRY', 'docker.io') . '/library';
+    my $os_reg        = 'registry.opensuse.org/opensuse';
+    my @docker_images = ("$ex_reg/alpine", "$os_reg/leap", "$os_reg/tumbleweed", "$ex_reg/debian", "$ex_reg/ubuntu", "$ex_reg/centos", "$ex_reg/fedora");
+    my @podman_images = ("$ex_reg/alpine", "$os_reg/leap", "$os_reg/tumbleweed", "$ex_reg/debian", "$ex_reg/ubuntu", "$ex_reg/centos", "$ex_reg/fedora");
```
Comment (Vogtinator) [inline tests/containers/containers_3rd_party.pm:70]: Unrelated cleanup:

```suggestion
    my @podman_images = @docker_images;
```

Or just change it everywhere to just `@images`...

File: tests/containers/containers_3rd_party.pm [hunk 2]
```diff
@@ -64,8 +64,10 @@ sub run {
     my ($running_version, $sp, $host_distri) = get_os_release;
 
     # Define test images here
-    my @docker_images = ('alpine', 'opensuse/leap', 'opensuse/tumbleweed', 'debian', 'ubuntu', 'centos', 'fedora');
-    my @podman_images = ('alpine', 'opensuse/leap', 'opensuse/tumbleweed', 'debian', 'ubuntu', 'centos', 'fedora');
+    my $ex_reg        = get_var('REGISTRY', 'docker.io') . '/library';
+    my $os_reg        = 'registry.opensuse.org/opensuse';
```
Comment (Vogtinator) [inline tests/containers/containers_3rd_party.pm:68]: The registry is just `registry.opensuse.org`

File: tests/containers/containers_3rd_party.pm [hunk 3]
```diff
@@ -64,8 +64,10 @@ sub run {
     my ($running_version, $sp, $host_distri) = get_os_release;
 
     # Define test images here
-    my @docker_images = ('alpine', 'opensuse/leap', 'opensuse/tumbleweed', 'debian', 'ubuntu', 'centos', 'fedora');
-    my @podman_images = ('alpine', 'opensuse/leap', 'opensuse/tumbleweed', 'debian', 'ubuntu', 'centos', 'fedora');
+    my $ex_reg        = get_var('REGISTRY', 'docker.io') . '/library';
```
Comment (Vogtinator) [inline tests/containers/containers_3rd_party.pm:67]: Technically `library` is part of the image name, so I'd move the `library` part to the image list

File: tests/containers/registry.pm [hunk 4]
```diff
@@ -36,7 +36,7 @@ sub registry_push_pull {
     die 'Argument $image not provided!'   unless $image;
     die 'Argument $runtime not provided!' unless $runtime;
 
-    # Pull $image from default registry
+    # Pull $image from  registry
```
Comment (Vogtinator) [inline tests/containers/registry.pm:39]:
```suggestion
    # Pull $image
```

or just drop the comment

File: tests/containers/containers_3rd_party.pm (hunk 3 above)
Comment (jlausuch) [reply tests/containers/containers_3rd_party.pm:67]: Yeah, it was just to avoid some text later. Maybe renaming the variable name?

File: tests/containers/containers_3rd_party.pm (hunk 2 above)
Comment (jlausuch) [reply tests/containers/containers_3rd_party.pm:68]: Same as above, to avoid this in lines below.

File: tests/containers/containers_3rd_party.pm (hunk 3 above)
Comment (Vogtinator) [reply tests/containers/containers_3rd_party.pm:67]: That would work too. `$os_prefix`/`$tp_prefix` maybe ("third-party")?

File: tests/containers/containers_3rd_party.pm (hunk 1 above)
Comment (jlausuch) [reply tests/containers/containers_3rd_party.pm]: done

File: tests/containers/registry.pm (hunk 4 above)
Comment (jlausuch) [reply tests/containers/registry.pm]: done

**Comment by jlausuch** [comment]
New VR: https://openqa.opensuse.org/tests/1492276

**Comment by jlausuch** [comment]
Green test after latest changes: https://openqa.opensuse.org/tests/1492285

**Review by Vogtinator — APPROVED** [review]
